config-model/src/main/java/com/yahoo/searchdefinition/derived/AttributeFields.java

```diff
@@ -134,6 +134,11 @@ private void deriveAttributeAsArrayType(ImmutableSDField field) {
         }
         Attribute attribute = field.getAttributes().get(field.getName());
         if (attribute != null) {
+            Ranking ranking = field.getRanking();
```

Comment:
I suggest creating a helper function for this (e.g. applyRankingSettings() / considerApplyRankingSettings()) and use that in deriveAttribute() as well to ensure we handle this the same way in both code paths.